### PR TITLE
nv2a: Implement vertex attribute setters

### LINKED ISSUE #671: pgraph: Several inline buffer vertex attribute setters are not implemented
### Bug Description

Edit: See https://github.com/mborgerson/xemu/pull/672/files#r791195300 for additional setters that should be implemented.

While writing a test for #623 I found that xemu doesn't implement a handful of the vertex attribute setters [used around here](https://github.com/abaire/nxdk_pgraph_tests/blob/4e2776729ed6a3ea86b4eb15247bc9009ed7cff9/src/test_host.cpp#L493).

Specifically:
```
diffuse w/ 4 floats
nv2a: pgraph method (0): 0x97 -> 0x1550 (0x3f800000)
nv2a:     unhandled  (0x97 0x00001550)

normal w/ 3 floats
nv2a: pgraph method (0): 0x97 -> 0x1530 (0x0)
nv2a:     unhandled  (0x97 0x00001530)

specular w/ 4 floats
nv2a: pgraph method (0): 0x97 -> 0x1570 (0x2d002c)
nv2a:     unhandled  (0x97 0x00001570)

texcoord0 w/ 2 floats (and tc1, tc2, tc3)
nv2a: pgraph method (0): 0x97 -> 0x1590 (0x2e0036)
nv2a:     unhandled  (0x97 0x00001590)

```

### Expected Behavior

xemu should implement these setters in case some game makes use of them

### xemu Version

Latest master w/ #623 patched in, but that change does not remove any support.

Version: 0.6.2-54-g10b29fc45c
Branch:  updates_normal_inline_value
Commit:  10b29fc45c4f62f51fc44531349b61e5a152cc14
Date:    Mon Jan 24 08:47:18 PM UTC 2022

### System Information

OS: Kubuntu 21.04 - 5.11.0-38-generic
CPU: Core i7-6700K @ 4GHz
GPU: NVIDIA GeForce GTX 1070
GPU Driver: 4.6.0 NVIDIA 470.74

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by abaire**
Fixes #671 

[Tests here](https://github.com/abaire/nxdk_pgraph_tests/blob/main/src/tests/attribute_carryover_tests.cpp)
Expected HW behavior is [here](https://github.com/abaire/nxdk_pgraph_tests_golden_results/wiki/Results-Attrib_carryover#t-d00_00_10_10-dapng) but note that the bottom line / right hand triangle will not render correctly without #623.

### CODE REVIEW

File: hw/xbox/nv2a/nv2a_regs.h [hunk 1]
```diff
@@ -1040,6 +1040,13 @@
 #   define NV097_SET_LIGHT_LOCAL_POSITION                     0x0000105C
 #   define NV097_SET_LIGHT_LOCAL_ATTENUATION                  0x00001068
 #   define NV097_SET_VERTEX4F                                 0x00001518
+#   define NV097_SET_NORMAL                                   0x00001530
```
Comment (mborgerson) [inline hw/xbox/nv2a/nv2a_regs.h:1043]: There are other variants (eg integer types) which we should implement while we are at it, see https://github.com/freedesktop-unofficial-mirror/nouveau__mesa/blob/02afd45d3b2eccff5d566cdeb32b3211803bd500/src/mesa/drivers/dri/nouveau/nouveau_reg.h#L3029

File: hw/xbox/nv2a/pgraph.c [hunk 2]
```diff
@@ -2241,6 +2241,70 @@ DEF_METHOD_INC(NV097, SET_VERTEX4F)
     }
 }
 
+
+DEF_METHOD_INC(NV097, SET_NORMAL)
```
Comment (mborgerson) [inline hw/xbox/nv2a/pgraph.c:2250]: Let's reduce the copy paste here with a macro

File: hw/xbox/nv2a/nv2a_regs.h [hunk 3]
```diff
@@ -1040,6 +1040,13 @@
...
+#   define NV097_SET_DIFFUSE_COLOR4F                          0x00001550
+#   define NV097_SET_SPECULAR_COLOR4F                         0x00001570
+#   define NV097_SET_TEXTURE0                                 0x00001590
```
Comment (mborgerson) [inline hw/xbox/nv2a/nv2a_regs.h]: Naming is a bit confusing. This is actually setting texture coordinates, so something like SET_TEXCOORD0 would be appropriate

Comment (abaire) [reply hw/xbox/nv2a/nv2a_regs.h]: Done

File: hw/xbox/nv2a/pgraph.c (hunk 2 above)
Comment (abaire) [reply hw/xbox/nv2a/pgraph.c:2250]: Done

File: hw/xbox/nv2a/nv2a_regs.h (hunk 1 above)
Comment (abaire) [reply hw/xbox/nv2a/nv2a_regs.h:1043]: Mind if I do that in a followup? I will need to make a bunch of changes to my test suite to exercise those paths.

Comment (mborgerson) [reply hw/xbox/nv2a/nv2a_regs.h:1043]: Sure, no problem

**Pull request merged**